File: ShoppingList/app/src/main/java/com/example/shoppinglist/MainActivity.java
package com.example.shoppinglist;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.Button;
import android.widget.EditText;

import com.google.android.material.snackbar.BaseTransientBottomBar;
import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {

    public static final String MESSAGE_KEY="Hello ";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button bton= (Button) findViewById(R.id.loginBtn);
        bton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EditText editTxtname = (EditText) findViewById(R.id.username);
                EditText editPwname = (EditText) findViewById(R.id.password);
                //if (editTxtname.getText().toString().equals("mal") && editPwname.getText().toString().equals("gg")) {
                if (editTxtname.getText().toString().length() >= 6 && editPwname.getText().toString().length() >= 6 && !editPwname.getText().toString().contains(" ")) {
                    Intent intent = new Intent(MainActivity.this, Shopping.class);
                    String name = editTxtname.getText().toString();
                    intent.putExtra(MESSAGE_KEY, name);
                    startActivity(intent);
                } else {
                    Snackbar mySnackbar = Snackbar.make(v, "Wrong credentials", BaseTransientBottomBar.LENGTH_SHORT);
                    mySnackbar.show();
                    EditText editTxtname1 = (EditText) findViewById(R.id.username);
                    editTxtname1.onEditorAction(EditorInfo.IME_ACTION_DONE);
                }
            }});
    }
}

File: SoftwarePatternsCA2/app/src/main/java/com/example/softwarepatternsca2/MainActivity.kt
package com.example.softwarepatternsca2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.example.softwarepatternsca2.databinding.ActivityMainBinding
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private var auth: FirebaseAuth? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        auth = FirebaseAuth.getInstance()

        binding.viewItemsBtn.setOnClickListener {
            startActivity(Intent(applicationContext, ViewItemsActivity::class.java))
        }

        binding.logOutBtn.setOnClickListener {
            auth?.signOut()
            startActivity(Intent(applicationContext, LoginActivity::class.java))
            Toast.makeText(applicationContext, "User has been logged out", Toast.LENGTH_SHORT).show()
        }
    }
}